packages/compass-web/sandbox/atlas-cluster-connections.tsx

```diff
@@ -0,0 +1,110 @@
+import { Banner, Button, Label } from '@mongodb-js/compass-components';
+import type { ConnectionInfo } from '@mongodb-js/connection-storage/renderer';
+import React, { useCallback, useState } from 'react';
+import { ConnectionsList } from './connections-list';
+
+type NdsCluster = {
+  uniqueId: string;
+  name: string;
+  srvAddress: string;
+};
+
+export function useAtlasClusterConnectionsList() {
+  const [signInStatus, setSignInStatus] = useState<
+    'initial' | 'in-progress' | 'success' | 'error'
+  >('initial');
+  const [signInError, setSignInError] = useState<string | null>(null);
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [groupId, setGroupId] = useState<string | null>(null);
+  const [connections, setConnections] = useState<ConnectionInfo[]>([]);
+
+  const signIn = useCallback(async () => {
+    try {
+      setSignInStatus('in-progress');
+      const { groupId } = await fetch('/authenticate', { method: 'POST' }).then(
+        (res) => {
+          return res.json() as Promise<{ groupId: string }>;
+        }
+      );
+      setGroupId(groupId);
+      const clusters = await fetch(
+        `/cloud-mongodb-com/nds/clusters/${groupId}`
+      ).then((res) => {
+        return res.json() as Promise<NdsCluster[]>;
+      });
+      setConnections(
+        clusters.map((cluster: NdsCluster): ConnectionInfo => {
+          return {
+            id: cluster.uniqueId,
+            connectionOptions: {
+              connectionString: `mongodb+srv://<username>:<password>@${cluster.srvAddress}`,
+            },
+            atlasMetadata: {
+              orgId: '',
+              projectId: groupId,
+              clusterId: cluster.uniqueId,
+              clusterName: cluster.name,
+              clusterType: 'replicaSet',
+              regionalBaseUrl: '',
+            },
+          };
+        })
+      );
+      setSignInStatus('success');
+    } catch (err) {
+      setSignInError((err as Error).message);
+      setSignInStatus('error');
+    }
+  }, []);
+
+  return { signIn, signInStatus, signInError, groupId, connections };
+}
+
+export function AtlasClusterConnectionsList({
+  connections,
+  onConnectionClick,
+  onConnectionDoubleClick,
+  signInStatus,
+  signInError,
+  onSignInClick,
+}: {
+  connections: ConnectionInfo[];
+  onConnectionClick(connectionInfo: ConnectionInfo): void;
+  onConnectionDoubleClick(connectionInfo: ConnectionInfo): void;
+  signInStatus: string;
+  signInError: string | null;
+  onSignInClick(): void;
+}) {
+  if (signInStatus === 'initial' || signInStatus === 'in-progress') {
+    return (
+      <Button
+        variant="primary"
+        type="button"
+        isLoading={signInStatus === 'in-progress'}
+        loadingText="Logging in..."
+        onClick={onSignInClick}
+      >
+        Log in to Atlas
+      </Button>
+    );
+  }
+
+  if (signInStatus === 'error' && signInError) {
```

Comment:
Probably the signin status could be just a typed object to avoid having issues where signInError is undefined or null but the status is error.

Something like:

`type SigninStatus = { status: 'error', error: string } | { status: 'in-progress' } ...`

Reply:
Makes sense, updated in https://github.com/mongodb-js/compass/pull/5634/commits/b233ffd1cd69e8706b57957c5bbc1af18670e7a1